spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.deployer.spi.app;
+
+import java.util.Map;
+
+/**
+ * Extension of the AppDeployer interface that adds an additional
+ * method to return the DeploymentState for a collection of deployment ids.
+ */
+public interface MultiStateAppDeployer extends AppDeployer {
```

Comment:
should go ahead and add `@Deprecated` with a mention that the method will move in a future release (likely 2.0)